Charm/Idle/IdleDetector.cpp

```diff
@@ -54,9 +57,11 @@ IdleDetector *IdleDetector::createIdleDetector(QObject *parent)
 #endif
 
 #ifdef CHARM_IDLE_DETECTION_AVAILABLE
-    X11IdleDetector *detector = new X11IdleDetector(parent);
-    detector->setAvailable(detector->idleCheckPossible());
-    return detector;
+    if (QX11Info::isPlatformX11()) {
```

Comment:
At least on Fedora 33, this returns true also for Wayland, thus the code is still executed and crashes.

Reply:
KWindowSystem uses

    if (QGuiApplication::platformName() == QLatin1String("xcb"))

do detect "real X11 without wayland"

Reply:
So does QX11Info::isPlatformX11(). https://code.woboq.org/qt5/qtx11extras/src/x11extras/qx11info_x11.cpp.html#_ZN8QX11Info13isPlatformX11Ev

Either way, it seems like more proper way to fix this crash will be to check if the Screen Saver extension is available.

Reply:
I've pushed a fix now, doing exactly that (checking for the extension)

Reply:
Great, thanks! I guess this MR can be closed now. :)

Reply:
Glad it's some sort of "alive" still. Thanks.